src/rdrand.rs

```diff
@@ -121,3 +121,23 @@ unsafe fn rdrand_exact(dest: &mut [MaybeUninit<u8>]) -> Option<()> {
     }
     Some(())
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{tests::*, *};
+
+    fn rdrand_fill_uninit(dest: &mut [MaybeUninit<u8>]) -> Result<&mut [u8], Error> {
+        super::getrandom_inner(dest)?;
+        Ok(unsafe { slice_assume_init_mut(dest) })
+    }
+
+    #[test]
```

Comment:
I support the general approach, I don't think it makes sense to manually repeat these tests in each module. We should have something that ensures the full suite of tests (more-or-less) is run for each implementation.

I have done this in other projects:
```
macro_rules! define_tests {
    ( $fill_init:path, ... ) => {
        #[test]
        fn test_small() { 
            tests::test_small($fill_uninit);
        }
       #[test]
      fn test_empty() { tests::test_empty($fill_uniit) }
   }
}
```
Then in each implementation (rdrand.rs, use_file.rs, etc.):
```
mod tests;
tests::define_tests!(rdrand_fill_uninit)

fn fill_uninit(dest: &mut [MaybeUninit<u8>]) -> Result<&mut [u8], Error> { ... }
```

Otherwise, we'd have to decide for each implementation whether we should copy/paste code into it, and naturally we'd gravitate towards less comprehensive testing of each implementation, which is the opposite of what we want--we want more comprehensive testing to be easier, not harder.

Reply:
Nice, I like this. Added a `define_tests` macro which makes reusing the tests much easier (see #476 #475 #474).